cmake/scripts/common/AddonHelpers.cmake

```diff
@@ -35,7 +35,81 @@ endmacro()
 # Build, link and optionally package an add-on
 macro (build_addon target prefix libs)
   addon_version(${target} ${prefix})
+
   if(${prefix}_SOURCES)
+    # Read used headers from addon, needed to identitfy used kodi addon interface headers
+    if(${prefix}_HEADERS)
+      # Add the used header files defined with CMakeLists.txt from addon itself
+      foreach(hdr_file ${${prefix}_HEADERS})
+        list(APPEND USED_SOURCES ${PROJECT_SOURCE_DIR}/${hdr_file})
+      endforeach()
+    else()
+      message(AUTHOR_WARNING "You have not defined used headers in cmake, scanning all addon included files!")
+      file(GLOB_RECURSE USED_SOURCES ${PROJECT_SOURCE_DIR}/*.h*)
+    endif()
+    # Add the used source files defined with CMakeLists.txt from addon itself
+    foreach(src_file ${${prefix}_SOURCES})
+      list(APPEND USED_SOURCES ${PROJECT_SOURCE_DIR}/${src_file})
+    endforeach()
+
+    # Set defines used in addon.xml.in and read from versions.h to set add-on
+    # version parts automatically
+    file(STRINGS ${KODI_INCLUDE_DIR}/versions.h BIN_ADDON_PARTS)
+    foreach(loop_var ${BIN_ADDON_PARTS})
+      # Only pass strings with "#define" from versions.h
+      if(loop_var MATCHES "#define")
```

Comment:
The regex is still Greek to me! :-D 
Has tried several strings and not found the right to become only the parts with "#define".